MetaLamp/lending-pool/src/Plutus/Contracts/Endpoints.hs

```diff
@@ -0,0 +1,225 @@
+
+import           Control.Monad                    hiding (fmap)
+import qualified Data.ByteString                  as BS
+import qualified Data.Map                         as Map
+import           Data.Monoid                      (Last (..))
+import           Data.Proxy                       (Proxy (..))
+import           Data.Text                        (Text, pack)
+import           Data.Void                        (Void)
+import           Ledger                           hiding (singleton)
+import           Ledger.Constraints               as Constraints
+import           Ledger.Constraints.OnChain       as Constraints
+import           Ledger.Constraints.TxConstraints as Constraints
+import qualified Ledger.Scripts                   as Scripts
+import qualified Ledger.Typed.Scripts             as Scripts
+import           Playground.Contract
+import           Plutus.Contract                  hiding (when)
+import qualified Plutus.Contracts.AToken          as AToken
+import           Plutus.Contracts.Core            (Aave, AaveDatum (..),
+                                                   AaveRedeemer (..),
+                                                   Reserve (..), ReserveId,
+                                                   UserConfig (..))
+import qualified Plutus.Contracts.Core            as Core
+import           Plutus.Contracts.Currency        as Currency
+import qualified Plutus.Contracts.FungibleToken   as FungibleToken
+import qualified Plutus.Contracts.State           as State
+import           Plutus.State.Select              (StateOutput (..))
+import           Plutus.V1.Ledger.Ada             (adaValueOf, lovelaceValueOf)
+import qualified Plutus.V1.Ledger.Address         as Addr
+import           Plutus.V1.Ledger.Value           as Value
+import qualified PlutusTx
+import           PlutusTx.Prelude                 hiding (Semigroup (..),
+                                                   unless)
+import           Prelude                          (Semigroup (..))
+import qualified Prelude
+import           Text.Printf                      (printf)
+
+newtype CreateParams =
+    CreateParams
+        { cpAsset :: AssetClass }
+    deriving (Show, Generic)
+    deriving anyclass (FromJSON, ToJSON, ToSchema)
+
+PlutusTx.makeLift ''CreateParams
+
+createReserve :: CreateParams -> Reserve
+createReserve CreateParams {..} =
+    Reserve
+        { rCurrency = cpAsset,
+          rAmount = 0,
+          rAToken = AToken.makeAToken cpAsset,
+          rDebtToken = cpAsset,
+          rLiquidityIndex = 1 }
+
+start :: HasBlockchainActions s => [CreateParams] -> Contract w s Text Aave
+start params = do
+    pkh <- pubKeyHash <$> ownPubKey
+    aaveToken  <- fmap Currency.currencySymbol $
+           mapError (pack . show @Currency.CurrencyError) $
+           Currency.forgeContract pkh [(Core.aaveProtocolName, 1)]
+    let aave = Core.aave aaveToken
+    let rootToken = assetClass aaveToken Core.aaveProtocolName
```

Comment:
seems like these lines are doing the same thing, can we reduce duplication and just unpack `Aave` after construction?
```
aave protocol = Aave (assetClass protocol aaveProtocolName)
```